client/auth.go

```diff
@@ -96,10 +96,17 @@ func (client *Client) InjectAuthenticationHeader(req *http.Request, path string)
 			Name:  "APIC-Certificate-Fingerprint",
 			Value: "fingerprint",
 		})
-		req.AddCookie(&http.Cookie{
+		if client.appUserName != "" {
+		   req.AddCookie(&http.Cookie{
+			Name:  "APIC-Certificate-DN",
+			Value: fmt.Sprintf("uni/userext/appuser-%s/usercert-%s", client.appUserName, client.appUserName),
```

Comment:
With the first comment, if you pass in both appUserName and adminCert, the only different would be the dn for the appuserName.  That would make this more flexible when you actually need to pass in different usercert names for the appUser.

Reply:
agreed.